### DISCUSSION TITLE
Is Gunicorn server supported to deploy FastAPI on Windows Server 2022 as Production Server

### DISCUSSION THREAD

**Discussion by technqvi** [fastapi/fastapi#6383] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
gunicorn -w 4 -k uvicorn.workers.UvicornWorker
```

### Description

I want to develop FastAPI on Windows sever 2022.
As I do some research on [https://www.uvicorn.org/deployment/](https://www.uvicorn.org/deployment/).

Is it possible for me to run this command   ` gunicorn -w 4 -k uvicorn.workers.UvicornWorker `  to run the production server on window sever 2022.
I don't have experience in it so I don't know to get started with it. 
 **_**I have 4 question_**
1. Gunicorn server supported to deploy FastAPI on Windows Server 2022 ?
2. Howto set script file to run ( you can guide me to article relevant to this issue) , I can study it further
3. If it doesn't support Windows Server, It is more likely to have 2 way to get it done.
  3.1 I install docker on the window server and deploy FastApo App to docker. (Let me know where I can download the image file)
  3.2. Can I apply IIS Server for FastAPI on Windows?**

I hardly find any content that literally explains step-by-step how to deploy the FastAPI App to  the Windows server.

On My Window server 2020 that I am working on , I deploy Django App  on IIS Server as The first application , so that's why I have to  The FastApi App as second Applitoin on this window server as well.

### Operating System

Windows

### Operating System Details

Edition	**Windows Server 2022 Datacenter Azure Edition**
Version	21H2
Installed on	‎2/‎11/‎2022
OS build	20348.643

Processor	Intel(R) Xeon(R) CPU E5-2673 v3 @ 2.40GHz   2.40 GHz
Installed RAM	7.99 GB
System type	64-bit operating system, x64-based processor
Pen and touch	Pen and touch support with 3 touch points

### FastAPI Version

fastapi 0.75.2

### Python Version

python 3.9

**Comment by Kludex** [comment, marked as answer]
Gunicorn doesn't support Windows.

**Comment by technqvi** [comment]
> Gunicorn doesn't support Windows.
Is there other ways to deploy fast API on window server?

**Comment by jgould22** [comment]
There are posts in the Gunicorn repo on tweaking things to get it running in windows linux subsystem but that is probably difficult and buggy way to do things.

In 2022 Windows is not a popular OS to run new server infrastructure and IIS is even less popular since they are basically the home of legacy C#/Java enterprise applications. Thus it is doubtful you will find a full featured walk through on how to use IIS with an ASGI application.

You best bet if you absolutely must run it in Windows Server is docker. There is a performance penalty because I think windows docker runs things in a VM but it will be the easiest. Here are the docs for docker https://fastapi.tiangolo.com/deployment/docker/
If you use docker it will also answer your question 2 since the start command will be covered by the docker file and you will simply invoke it as any other docker container.

**Comment by technqvi** [comment]
> There are posts in the Gunicorn repo on tweaking things to get it running in windows linux subsystem but that is probably difficult and buggy way to do things.
> 
> In 2022 Windows is not a popular OS to run new server infrastructure and IIS is even less popular since they are basically the home of legacy C#/Java enterprise applications. Thus it is doubtful you will find a full featured walk through on how to use IIS with an ASGI application.
> 
> You best bet if you absolutely must run it in Windows Server is docker. There is a performance penalty because I think windows docker runs things in a VM but it will be the easiest. Here are the docs for docker https://fastapi.tiangolo.com/deployment/docker/ If you use docker it will also answer your question 2 since the start command will be covered by the docker file and you will simply invoke it as any other docker container.

it is understandable.
Thank you for your recommendation

**Comment by mgzon** [comment]
Hi @technqvi 
Just wondering if you proceeded with the docker solution as suggested above? I'm currently looking for a solution on this as well and just would like to check if you found the above stable to use in production or not.
